syntax = "proto3";

package discovery_service;

// Discovery Service
service Registration {
    rpc RegisterService(RegisterServiceRequest) returns (RegisterServiceResponse) {}
    rpc DeregisterService(DeregisterServiceRequest) returns (DeregisterServiceResponse) {}
}

service Retrieval {
    rpc GetService(GetServiceRequest) returns (GetServiceResponse) {}
}

service Health {
    rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse) {}
}

// Register Server
message RegisterServiceRequest {
    string service_type = 1;
    string service_address = 2;
    string service_port = 3;
}

message RegisterServiceResponse {
    string service_id = 1;
}

// De-register Server
message DeregisterServiceRequest {
    string service_id = 1;
}

message DeregisterServiceResponse {
    bool success = 1;
}

// Get Service
message GetServiceRequest {
    string service_type = 1;
}

message GetServiceResponse {
    string service_host = 1;
    string service_port = 2;
}

// Health Check
message HealthCheckRequest {
    string serviceName = 1;
}

message HealthCheckResponse {
    bool healthy = 1;
}

